dask/dataframe/core.py

```diff
@@ -4523,6 +4524,9 @@ def __setattr__(self, key, value):
     def __getattr__(self, key):
         if key in self.columns:
             return self[key]
+        elif key == "empty":
+            # To raise informative exception
+            object.__getattribute__(self, key)
```

Comment:
@pavithraes and I had to puzzle through the control flow here, which isn't very obvious:

`df.empty` --> `DataFrame.__getattribute__(df, "empty")` --> `AttributeError` --> `DataFrame.__getattr__(df, "empty")` --> `object.__getattribute__(df, "empty")`

It's a bit convoluted, but I think probably okay. Would you mind adding a more informative comment for our future selves?

Reply:
Ha, yeah, it is a bit convoluted.  Done.  Thanks for looking and puzzling through it!